docs/examples/legacy/Legacy.ipynb

```diff
@@ -0,0 +1,127 @@
+    "\n",
+    "One DataArray can only be part of at most one DataSet. This ensures that we don’t generate irreversible situations by saving an array in multiple places and reloading them separately, or conflicts if we try to sync (or reload) several DataSets with inconsistent data in the multiply-referenced arrays, and that we can always refer from the DataArray to a single DataSet, which is important for live plotting.\n",
+    "\n",
+    "The DataSet also specifies where and how it is to be stored on disk. Storage is specified by an io_manager (the physical device / protocol, and base location in the normal case of disk storage), a location (string, relative path within the io manager), and formatter (specifies the file type and how to read to and write from a DataSet).\n",
+    "\n",
+    "### Responsibilities:\n",
+    "\n",
+    "Accepting incremental pieces of data (setpoints and measurements as they become available)\n",
+    "\n",
+    "Either holding that data locally (within its DataArrays), or pushing it to another copy of itself that stores it\n",
+    "\n",
+    "If it’s a copy that holds data, each DataArray maintains a record of the range of indices that have changed since the last save to storage, the last index that has been saved, and (if it’s in PULL_FROM_SERVER mode) the last index that has been synced from the server. This implicitly assumes that the DataArrays are filled in order of the raveled indices, ie looping over the inner index first.\n",
+    "\n",
+    "It’s up to the Formatter to look at each of these DataArrays, decide what parts of the changes in each to save to storage, and then tell each DataArray what it saved (expressed as a greatest raveled index). With that information the DataArray updates its record of what still needs saving. This is done so that a Formatter can choose to combine several DataArrays into one table, which may require writing only the values at array positions which have been finished in all of these arrays.\n",
+    "\n",
+    "Each DataSet holds:\n",
+    "Its own metadata (JSON-compatible dict, i.e., everything that the custom JSON encoder class qcodes.utils.helpers.NumpyJSONEncoder supports.)\n",
+    "\n",
+    "Its mode (PUSH_TO_SERVER, PULL_FROM_SERVER, LOCAL)\n",
+    "\n",
+    "A dict of DataArrays, each with attributes: name (which is also its dictionary key in DataSet.arrays), label, units, setpoints. If the DataSet is in PUSH_TO_SERVER mode, these DataArrays do not hold any data. Otherwise, these DataArrays contain numpy arrays of data, as well as records (as described above) of what parts of that array have been changed, saved, and synced.\n",
+    "\n",
+    "location, formatter, and io manager\n",
+    "\n",
```

Comment:
```suggestion
    "Each DataSet holds:\n",
    "\n",
    "- Its own metadata (JSON-compatible dict, i.e., everything that the custom JSON encoder class qcodes.utils.helpers.NumpyJSONEncoder supports.)\n",
    "- Its mode (PUSH_TO_SERVER, PULL_FROM_SERVER, LOCAL)\n",
    "- A dict of DataArrays, each with attributes: name (which is also its dictionary key in DataSet.arrays), label, units, setpoints. If the DataSet is in PUSH_TO_SERVER mode, these DataArrays do not hold any data. Otherwise, these DataArrays contain numpy arrays of data, as well as records (as described above) of what parts of that array have been changed, saved, and synced.\n",
    "- location, formatter, and io manager\n",
    "\n",
```